woocommerce/woocommerce
# Create attribute within the new product MVP

### All Submissions:






### Changes proposed in this Pull Request:

This allows users to create a new attribute if the name they are searching for doesn't exist within the attribute list.
This is currently dependent on #34999 to be merged, which is why it's pointing at that branch.

https://user-images.githubusercontent.com/2240960/195870763-a609958f-b3ab-4515-a498-25660423b5c9.mp4

Address the creation part in #34331  .

### How to test the changes in this Pull Request:

1. Load this branch, build it, and make sure you have the `new-product-management-experience` feature flag enabled (you can do so using the latest version of the Beta tester within the mono repo).
2. Go to **Products > Add new (MVP)** and fill out some of the initial details like name and price
3. Scroll down and click **Add first attribute** in the attributes section
4. Now write a random name in the attribute search field in the table, of an attribute that doesn't exist.
5. The 'Create "your new attribute name"' should show as an option
6. Click that and the new item should show as selected and the term field should focus
7. With `localStorage.setItem( 'debug', 'wc-admin:*' );` run prior, it should show that a `product_attribute_add_custom_attribute` has been fired.
8. Start adding attribute terms, this dropdown should initial be blank and any text you put in will show up as a `Create "text"` option, allowing you to add several new items.
9. Click **Add** the custom attribute should show correctly in the main list, now hit `Update` or `Publish`
10. Go to old product editor and load the same product you just saved, check the attributes tab and make sure the custom one's show up correctly.

### Other information:




### FOR PR REVIEWER ONLY:
